syntax = "proto3";

service AnomaliesService{
  rpc CreateRecordAnomalies (CreateAnomaliesRequest) returns (CreateAnomaliesResponse);
  rpc ReadRecordAnomalies (ReadAnomaliesRequest) returns (ReadAnomaliesResponse);
  rpc UpdateRecordAnomalies(UpdateAnomaliesRequest) returns (UpdateAnomaliesResponse);
  rpc DeleteRecordAnomalies (DeleteAnomaliesRequest) returns (DeleteAnomaliesResponse);
}

message CreateAnomaliesRequest {
  int64 projectId = 1;
  bytes data = 2;
  int32 status = 3;
  string name = 4;
  string tags = 5;
  string description = 6;
  int64 radius = 7;
  int64 scale = 8;
  int64 proccessedByMemberId = 9;
}

message CreateAnomaliesResponse {
  bool success = 1;
  repeated string message = 2;
}


message ReadAnomaliesRequest {
  int64 id = 1;
  int64 projectId = 2;
  bytes data = 3;
  string timestamp = 4;
  int32 status = 5;
  string name = 6;
  string tags = 7;
  string description = 8;
  int64 radius = 9;
  int64 scale = 10;
  int64 proccessedByMemberId = 11;
}

message ReadAnomaliesResponse {
  bool success = 1;
  repeated string message = 2;
  repeated AnomaliesObject data = 3;
}

message AnomaliesObject{
  int64 id = 1;
  int64 projectId = 2;
  bytes data = 3;
  string timestamp = 4;
  int32 status = 5;
  string name = 6;
  string tags = 7;
  string description = 8;
  int64 radius = 9;
  int64 scale = 10;
  int64 proccessedByMemberId = 11;
}


message UpdateAnomaliesRequest {
  repeated int64 id = 1;
  repeated int64 projectId = 2;
  repeated bytes data = 3;
  repeated string timestamp = 4;
  repeated int32 status = 5;
  repeated string name = 6;
  repeated string tags = 7;
  repeated string description = 8;
  repeated int64 radius = 9;
  repeated int64 scale = 10;
  repeated int64 proccessedByMemberId = 11;
  UpdateAnomaliesObject update_data = 12;
}

message UpdateAnomaliesObject{
  int64 projectId = 1;
  bytes data = 2;
  string timestamp = 3;
  int32 status = 4;
  string name = 5;
  string tags = 6;
  string description = 7;
  int64 radius = 8;
  int64 scale = 9;
  int64 proccessedByMemberId = 10;
}

message UpdateAnomaliesResponse {
  bool success = 1;
  repeated string message = 2;
}

message DeleteAnomaliesRequest {
  repeated int64 id = 1;
  repeated int64 projectId = 2;
  repeated bytes data = 3;
  repeated string timestamp = 4;
  repeated int32 status = 5;
  repeated string name = 6;
  repeated string tags = 7;
  repeated string description = 8;
  repeated int64 radius = 9;
  repeated int64 scale = 10;
  repeated int64 proccessedByMemberId = 11;
}

message DeleteAnomaliesResponse {  
  bool success = 1;
  repeated string message = 2;
}